JavaCPP fails tests on ARM64

Trying to build on ARM64

Note the x86_64 in the error log ...
Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so
Below

Note (2) a quick look around reveals libjniCalc.so on...
./target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/linux-arm64/libjniCalc.so

Edited src/it/osgi/pom.xml and changed the path - build worked (hackety, hack, hack(

===== Log =====

[INFO] [INFO] ------------------< org.bytedeco:javacpp-tests-osgi >-------------------
[INFO] [INFO] Building JavaCPP OSGi Tests 1.5.2-SNAPSHOT
[INFO] [INFO] --------------------------------[ jar ]---------------------------------
[INFO] [INFO]
[INFO] [INFO] --- javacpp:1.5.2-SNAPSHOT:build (javacpp.parser) @ javacpp-tests-osgi ---
[INFO] [WARNING] No classes found in package org.bytedeco.javacpp.test.osgi
[INFO] [INFO] Detected platform "linux-arm64"
[INFO] [INFO] Building for platform "linux-arm64"
[INFO] [INFO]
[INFO] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ javacpp-tests-osgi ---
[INFO] [WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] [INFO] skip non existing resourceDirectory /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/src/main/resources
[INFO] [INFO]
[INFO] [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ javacpp-tests-osgi ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] [INFO] Compiling 2 source files to /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes
[INFO] [INFO]
[INFO] [INFO] --- javacpp:1.5.2-SNAPSHOT:build (javacpp.compiler) @ javacpp-tests-osgi ---
[INFO] [INFO] Detected platform "linux-arm64"
[INFO] [INFO] Building for platform "linux-arm64"
[INFO] [WARNING] Could not load platform properties for class org.bytedeco.javacpp.test.osgi.JavaCPPOsgiTest
[INFO] [INFO] Generating /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp
[INFO] [INFO] Generating /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp
[INFO] [INFO] Compiling /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/linux-arm64/libjniCalc.so
[INFO] [INFO] g++ -I/home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/src/main/c -I/usr/lib/jvm/java-8-openjdk-arm64/include -I/usr/lib/jvm/java-8-openjdk-arm64/include/linux /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp -O3 -s -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniCalc.so
[INFO] [INFO] Deleting /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp
[INFO] [INFO] Deleting /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp
[INFO] [INFO]
[INFO] [INFO] --- bnd-maven-plugin:4.2.0:bnd-process (default) @ javacpp-tests-osgi ---
[INFO] [ERROR] /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/pom.xml [0:0]: Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  11.117 s
[INFO] [INFO] Finished at: 2019-10-03T16:19:18+01:00
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [ERROR] Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:4.2.0:bnd-process (default) on project javacpp-tests-osgi: Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so -> [Help 1]
[INFO] [ERROR]
[INFO] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[INFO] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[INFO] [ERROR]
[INFO] [ERROR] For more information about the errors and possible solutions, please read the following articles:
[INFO] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[INFO]   The build exited with code 1. See /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/build.log for details.
[INFO]           osgi/pom.xml ..................................... FAILED (15.2 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 0, Failed: 1, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  osgi/pom.xml
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:07 min
[INFO] Finished at: 2019-10-03T16:19:18+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.2.0:run (integration-test) on project javacpp: 1 build failed. See console output above for details. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

OK - educate me - src/it/osgi/pom.xml is unaltered I note

Both Nano + Pi die in different ways (Nano I know how to fix)

Am I meant to be running off a branch rather than master?

Does the version of Java matter? Got 8 on Nano, 11 on Pi

The Pi log (Eek)
================================
pi@raspberrypi:~/src/jcpp/javacpp $ mvn clean install 2>&1 | tee debug.log
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO]
[INFO] ------------------------< org.bytedeco:javacpp >------------------------
[INFO] Building JavaCPP 1.5.2-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ javacpp ---
[INFO] Deleting /home/pi/src/jcpp/javacpp/target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ javacpp ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 34 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ javacpp ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 138 source files to /home/pi/src/jcpp/javacpp/target/classes
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[25,16] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[25,16] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[25,16] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[34,28] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[34,28] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[34,28] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[37,9] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[WARNING] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java:[51,18] sun.misc.Unsafe is internal proprietary API and may be removed in a future release
[INFO] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/ClassProperties.java: Some input files use or override a deprecated API.
[INFO] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/ClassProperties.java: Recompile with -Xlint:deprecation for details.
[INFO] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/Pointer.java: Some input files use unchecked or unsafe operations.
[INFO] /home/pi/src/jcpp/javacpp/src/main/java/org/bytedeco/javacpp/Pointer.java: Recompile with -Xlint:unchecked for details.
[INFO]
[INFO] --- maven-plugin-plugin:3.5.2:descriptor (default-descriptor) @ javacpp ---
[INFO] Using 'UTF-8' encoding to read mojo source files.
[INFO] java-javadoc mojo extractor found 0 mojo descriptor.
[INFO] java-annotations mojo extractor found 2 mojo descriptors.
[INFO]
[INFO] --- bnd-maven-plugin:4.2.0:bnd-process (default) @ javacpp ---
[WARNING] Cannot consider developer in line '32' of file '/home/pi/src/jcpp/javacpp/pom.xml' for bundle header 'Bundle-Developers' as it does not contain the mandatory id.
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ javacpp ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ javacpp ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 6 source files to /home/pi/src/jcpp/javacpp/target/test-classes
[INFO] /home/pi/src/jcpp/javacpp/src/test/java/org/bytedeco/javacpp/PointerTest.java: Some input files use unchecked or unsafe operations.
[INFO] /home/pi/src/jcpp/javacpp/src/test/java/org/bytedeco/javacpp/PointerTest.java: Recompile with -Xlint:unchecked for details.
[INFO]
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ javacpp ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.bytedeco.javacpp.BufferTest
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBufferTest.cpp
Compiling /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/linux-armhf/libjniBufferTest.so
g++ -I/usr/lib/jvm/java-11-openjdk-armhf/include -I/usr/lib/jvm/java-11-openjdk-armhf/include/linux /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBufferTest.cpp /home/pi/src/jcpp/javacpp/jnijavacpp.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -std=c++11 -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniBufferTest.so
Deleting /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBufferTest.cpp
Deleting /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Loader
LongBuffer
ByteBuffer
BufferCallback
ShortBuffer
IntBuffer
FloatBuffer
DoubleBuffer
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.176 s - in org.bytedeco.javacpp.BufferTest
[INFO] Running org.bytedeco.javacpp.EnumTest
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniEnumTest.cpp
Compiling /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/linux-armhf/libjniEnumTest.so
g++ -I/usr/lib/jvm/java-11-openjdk-armhf/include -I/usr/lib/jvm/java-11-openjdk-armhf/include/linux /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniEnumTest.cpp /home/pi/src/jcpp/javacpp/jnijavacpp.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -std=c++11 -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniEnumTest.so
Deleting /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniEnumTest.cpp
Deleting /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Loader
Enum
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.007 s - in org.bytedeco.javacpp.EnumTest
[INFO] Running org.bytedeco.javacpp.BuilderTest
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Nothing generated for /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Compiling /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/linux-armhf-ext1/libjniBuilderTest.so
g++ -I/usr/lib/jvm/java-11-openjdk-armhf/include -I/usr/lib/jvm/java-11-openjdk-armhf/include/linux /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp /home/pi/src/jcpp/javacpp/jnijavacpp.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniBuilderTest.so
Deleting /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Deleting /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Loader
Note: UnsatisfiedLinkError should get thrown here and printed below.
java.lang.UnsatisfiedLinkError: /path/to/nowhere: cannot open shared object file: No such file or directory
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Compiling /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/linux-armhf-ext2/libjniBuilderTest.so
g++ -I/usr/lib/jvm/java-11-openjdk-armhf/include -I/usr/lib/jvm/java-11-openjdk-armhf/include/linux /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp /home/pi/src/jcpp/javacpp/jnijavacpp.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniBuilderTest.so
Deleting /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniBuilderTest.cpp
Deleting /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Loader
17
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 12.63 s - in org.bytedeco.javacpp.BuilderTest
[INFO] Running org.bytedeco.javacpp.PointerTest
Builder
Generating /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Generating /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniPointerTest.cpp
Compiling /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/linux-armhf-ext1/libjniPointerTest.so
g++ -I/usr/lib/jvm/java-11-openjdk-armhf/include -I/usr/lib/jvm/java-11-openjdk-armhf/include/linux /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniPointerTest.cpp /home/pi/src/jcpp/javacpp/jnijavacpp.cpp -march=armv6 -mfpu=vfp -mfloat-abi=hard -O3 -s -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniPointerTest.so
Deleting /home/pi/src/jcpp/javacpp/target/test-classes/org/bytedeco/javacpp/jniPointerTest.cpp
Deleting /home/pi/src/jcpp/javacpp/jnijavacpp.cpp
Loader
4 4 1
FloatPointer
Note: OutOfMemoryError should get thrown here and printed below.
java.lang.OutOfMemoryError: Cannot allocate new FloatPointer(33554433): totalBytes = 896M, physicalBytes = 32224K
java.lang.OutOfMemoryError: Failed to allocate memory within limits: totalBytes (896M + 128M) > maxBytes (1024M)
134283268 134217732
BytePointer
[WARNING] Corrupted STDOUT by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /home/pi/src/jcpp/javacpp/target/surefire-reports/2019-10-04T12-06-25_121-jvmRun1.dumpstream
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:18 min
[INFO] Finished at: 2019-10-04T12:06:58+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project javacpp: There are test failures.
[ERROR]
[ERROR] Please refer to /home/pi/src/jcpp/javacpp/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/pi/src/jcpp/javacpp && /usr/lib/jvm/java-11-openjdk-armhf/bin/java -jar /home/pi/src/jcpp/javacpp/target/surefire/surefirebooter9015031531813455552.jar /home/pi/src/jcpp/javacpp/target/surefire 2019-10-04T12-06-25_121-jvmRun1 surefire7961088032846095508tmp surefire_02471992091295902581tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR] Crashed tests:
[ERROR] org.bytedeco.javacpp.PointerTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/pi/src/jcpp/javacpp && /usr/lib/jvm/java-11-openjdk-armhf/bin/java -jar /home/pi/src/jcpp/javacpp/target/surefire/surefirebooter9015031531813455552.jar /home/pi/src/jcpp/javacpp/target/surefire 2019-10-04T12-06-25_121-jvmRun1 surefire7961088032846095508tmp surefire_02471992091295902581tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 134
[ERROR] Crashed tests:
[ERROR] org.bytedeco.javacpp.PointerTest
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
p

--- a/src/it/osgi/pom.xml
+++ b/src/it/osgi/pom.xml
@@ -14,6 +14,18 @@
   
   <profiles>
     <profile>
+      <id>Windows</id>
+      <activation>
+        <os>
+          <name>windows</name>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>windows-x86_64/libjniCalc.dylib</lib.name>
+        <os.name>Windows</os.name>
+      </properties>
+    </profile>
+    <profile>
       <id>macosx</id>
       <activation>
         <os>
@@ -26,14 +38,145 @@
       </properties>
     </profile>
     <profile>
-      <id>linux</id>
+      <id>linux-x86_64</id>
       <activation>
         <os>
           <name>Linux</name>
+          <arch>x86_64</arch>
         </os>
       </activation>
       <properties>
         <lib.name>linux-x86_64/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-x86-64</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>x86-64</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-x86_64/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-amd64</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>amd64</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-x86_64/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-x86</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>x86</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-x86/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-arm</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>arm</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-arm/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-armhf</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>armhf</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-armhf/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-arm64</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>aarch64</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-arm64/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-ppc</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>ppc</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-ppc/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-ppc64</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>ppc64</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-ppc64/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-ppc64le</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>ppc64le</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-ppc64le/libjniCalc.so</lib.name>
+        <os.name>Linux</os.name>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-mips64el</id>
+      <activation>
+        <os>
+          <name>Linux</name>
+          <arch>ppc64le</arch>
+        </os>
+      </activation>
+      <properties>
+        <lib.name>linux-mips64el/libjniCalc.so</lib.name>
         <os.name>Linux</os.name>
       </properties>
     </profile>

--- a/src/it/osgi/pom.xml
+++ b/src/it/osgi/pom.xml
@@ -10,174 +10,127 @@
   
   <properties>
     <bnd.version>4.2.0</bnd.version>
+    <lib.path>${os.name}-${os.arch}/${lib.name}</lib.path>
   </properties>
   
+  <!-- Copied and adapted from the javacpp-presets parent pom. Unfortunately 
+       inheriting these isn't appropriate because it would pull in other 
+       profiles which attempt to build sub-projects which don't exist.
+       Also, importing as a parent would introduce a dependency cycle between
+       javacpp and the javacpp-presets  -->
   <profiles>
     <profile>
-      <id>Windows</id>
+      <id>linux</id>
       <activation>
-        <os>
-          <name>windows</name>
-        </os>
+        <os><name>linux</name></os>
       </activation>
       <properties>
-        <lib.name>windows-x86_64/libjniCalc.dylib</lib.name>
-        <os.name>Windows</os.name>
+        <os.kernel>linux</os.kernel>
+        <os.name>linux</os.name>
+        <lib.name>libjniCalc.so</lib.name>
       </properties>
     </profile>
     <profile>
       <id>macosx</id>
       <activation>
-        <os>
-          <name>Mac OS X</name>
-        </os>
+        <os><name>mac os x</name></os>
       </activation>
       <properties>
-        <lib.name>macosx-x86_64/libjniCalc.dylib</lib.name>
-        <os.name>MacOSX</os.name>
+        <os.kernel>darwin</os.kernel>
+        <os.name>macosx</os.name>
+        <lib.name>libjniCalc.dylib</lib.name>
       </properties>
     </profile>
     <profile>
-      <id>linux-x86_64</id>
+      <id>windows</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>x86_64</arch>
-        </os>
+        <os><family>windows</family></os>
       </activation>
       <properties>
-        <lib.name>linux-x86_64/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.kernel>windows</os.kernel>
+        <os.name>windows</os.name>
+        <lib.name>jniCalc.dll</lib.name>
       </properties>
     </profile>
     <profile>
-      <id>linux-x86-64</id>
+      <id>arm</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>x86-64</arch>
-        </os>
+        <os><arch>arm</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-x86_64/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>armhf</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-amd64</id>
+      <id>aarch64</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>amd64</arch>
-        </os>
+        <os><arch>aarch64</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-x86_64/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>arm64</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-x86</id>
+      <id>armv8</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>x86</arch>
-        </os>
+        <os><arch>armv8</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-x86/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>arm64</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-arm</id>
+      <id>i386</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>arm</arch>
-        </os>
+        <os><arch>i386</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-arm/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-armhf</id>
+      <id>i486</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>armhf</arch>
-        </os>
+        <os><arch>i486</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-armhf/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-arm64</id>
+      <id>i586</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>aarch64</arch>
-        </os>
+        <os><arch>i586</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-arm64/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-ppc</id>
+      <id>i686</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>ppc</arch>
-        </os>
+        <os><arch>i686</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-ppc/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-ppc64</id>
+      <id>amd64</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>ppc64</arch>
-        </os>
+        <os><arch>amd64</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-ppc64/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86_64</os.arch>
       </properties>
     </profile>
     <profile>
-      <id>linux-ppc64le</id>
+      <id>x86-64</id>
       <activation>
-        <os>
-          <name>Linux</name>
-          <arch>ppc64le</arch>
-        </os>
+        <os><arch>x86-64</arch></os>
       </activation>
       <properties>
-        <lib.name>linux-ppc64le/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
-      </properties>
-    </profile>
-    <profile>
-      <id>linux-mips64el</id>
-      <activation>
-        <os>
-          <name>Linux</name>
-          <arch>ppc64le</arch>
-        </os>
-      </activation>
-      <properties>
-        <lib.name>linux-mips64el/libjniCalc.so</lib.name>
-        <os.name>Linux</os.name>
+        <os.arch>x86_64</os.arch>
       </properties>
     </profile>
   </profiles>
@@ -278,7 +231,7 @@
 Test-Cases: ${classes;NAMED;*Test}
 
 Bundle-NativeCode: \
- org/bytedeco/javacpp/test/osgi/${lib.name};\
+ org/bytedeco/javacpp/test/osgi/${lib.path};\
  osname=${os.name}
  
           ]]></bnd>

